tests/app/main/test_frameworks.py

```diff
@@ -3395,94 +3392,53 @@ def _send_email(self, clarification_question):
             data={'clarification_question': clarification_question}
         )
 
-    def _assert_clarification_email(self, mandrill_send_email, notify_send_email, is_called=True, succeeds=True,
-                                    clarification_question=''):
+    def _assert_clarification_email(self, notify_send_email, is_called=True, succeeds=True,
+                                    clarification_question='', times_called=0):
```

Comment:
Do we need all of `times_called`, `succeeds` and `is_called`? I'd recommend having something like 

```
assert notify_send_email.call_count == no_of_emails_sent
if no_of_emails_sent:
   assert notify_send_email.call_args_list == [
       mock.call(mock.ANY, to_email_address=...)
   ]
```

Reply:
I'd also keep the `_assert_clarification_email` and `_assert_framework_application_email` separate, then in the test we can call both 

```
# clarification email
self._assert_clarification_email(..., no_of_emails_sent=1)
self._assert_framework_application_email(..., no_of_emails_sent=0)
# framework application email
self._assert_clarification_email(..., no_of_emails_sent=0)
self._assert_framework_application_email(..., no_of_emails_sent=1)
```

Reply:
The `succeeds` keyword is useful for testing the behaviour under test in `test_should_be_a_503_if_email_fails`. In that case, we expect there to only be one call, because when it fails the whole view aborts and doesn't even try to send the second email. However, to test that behaviour in the same method I think you need to have
```
if not succeeds:
    notify_send_email.assert_has_calls(# email to DMP)
else:
    notify_send_email.assert_has_calls(email to DMP, receipt to supplier)
```

Reply:
I think the test scenarios themselves need a bit of a rethink. Let's chat when you have a sec.